win32ss/user/ntuser/nonclient.c

```diff
@@ -393,8 +393,13 @@ DefWndDoSizeMove(PWND pwnd, WORD wParam)
       if (!co_IntGetPeekMessage(&msg, 0, 0, 0, PM_REMOVE, TRUE)) break;
       if (IntCallMsgFilter( &msg, MSGF_SIZE )) continue;
 
-      /* Exit on button-up */
-      if (msg.message == WM_LBUTTONUP)
+      if (!g_bWindowSnapEnabled) /* If no WindowSnapEnabled: Exit on button-up, Return, or Esc */
+      {
+         if((msg.message == WM_LBUTTONUP) ||
```

Comment:
You can move the following code above here:
```c
      /* Exit on Return or Esc */
      if (msg.message == WM_KEYDOWN &&
          (msg.wParam == VK_RETURN || msg.wParam == VK_ESCAPE))
      {
         break;
      }
```

Reply:
Possible that I am just not clever enough to understand the code entirely, but are you really 100% sure that this would be safe? I mean we wouldn't call then any longer code like
```
pwnd->InternalPos.NormalRect = origRect;
co_IntSendMessage(UserHMGetHandle(pwnd), WM_SYSCOMMAND, SC_MAXIMIZE, 0);
```
and so on...
pwnd is a pointer from outside the func. So we may end up with a different state of that Window when exiting (which would most likely be the state of the previous iteration).
Are you 100% sure that this will not change the behavior?

Is it still guaranteed for the Window to properly get back to its original position?
Sorry for the question, but I really don't want to break inside here. :-)

Reply:
I can say that it depends `UserIsEnteredExclusive()` or not. If exclusive mode, the execution keeps the `PWND` data.

Reply:
It'll be better if we wrote `NT_ASSERT(UserIsEnteredExclusive());` in some places.

Reply:
You can increment the reference counter of `WND` by calling `UserRefObjectCo` function.

Reply:
Okay, we tested and worked in parallel on moving up the checks. And both came to the conclusion that it works.
Thank you for your input. And your idea for fusing the TRACEs.
Suggestion resolved? (see latest commits)